x/wasm/keeper/keeper.go

```diff
@@ -1213,3 +1218,37 @@ func (h DefaultWasmVMContractResponseHandler) Handle(ctx sdk.Context, contractAd
 	}
 	return result, nil
 }
+
+// PruneWasmCodes deletes code info for unpinned codes smaller than given code id.
+func (k Keeper) PruneWasmCodes(ctx sdk.Context, codeID uint64) error {
+	pinnedCodesHash := make(map[string]struct{}, 0)
+	unpinnedCodes := make([]uint64, 0)
+
+	for c := uint64(1); c < codeID; c++ {
+		info := k.GetCodeInfo(ctx, c)
+		if info == nil {
+			continue
+		}
+
+		if k.IsPinnedCode(ctx, c) {
+			pinnedCodesHash[hex.Dump(info.CodeHash)] = struct{}{}
+			continue
+		}
+		unpinnedCodes = append(unpinnedCodes, c)
+	}
```

Comment:
Could we loop over codes (or code infos) using a DB iterator instead of an integer? Then we don't need to execute the loop body so often when a lot of codes are already pruned.